### PR TITLE
Updates to CivModCore

### PR DESCRIPTION
**Pull request by Protonull**

### CODE REVIEW

File: src/main/java/vg/civcraft/mc/civmodcore/CivModCorePlugin.java [hunk 1]
```diff
@@ -1,28 +1,52 @@
...
 import vg.civcraft.mc.civmodcore.playersettings.gui.ConfigCommand;
+import vg.civcraft.mc.civmodcore.scoreboard.ScoreBoardListener;
 
 public class CivModCorePlugin extends ACivMod {
 
 	private static CivModCorePlugin instance;
 
 	@Override
-	public void onEnable() {
+	public synchronized void onEnable() {
```
Comment (Maxopoly) [inline src/main/java/vg/civcraft/mc/civmodcore/CivModCorePlugin.java]: Why is this synchronized. All calls to onEnable and onDisable will always come from the same thread

File: src/main/java/vg/civcraft/mc/civmodcore/itemHandling/NiceNames.java [hunk 2]
```diff
@@ -4,45 +4,20 @@
...
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
 import org.bukkit.Bukkit;
-import org.bukkit.Material;
 import org.bukkit.enchantments.Enchantment;
-import org.bukkit.inventory.ItemStack;
-import org.bukkit.inventory.meta.ItemMeta;
 
 public class NiceNames {
```
Comment (Maxopoly) [inline src/main/java/vg/civcraft/mc/civmodcore/itemHandling/NiceNames.java:14]: Should adjust the name of this class if its only holding aliases for Enchantments

File: src/main/java/vg/civcraft/mc/civmodcore/events/MaterialNamesLoadEvent.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package vg.civcraft.mc.civmodcore.events;
+
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+
+import javax.annotation.Nonnull;
+
+public final class MaterialNamesLoadEvent extends Event {
```
Comment (Maxopoly) [inline src/main/java/vg/civcraft/mc/civmodcore/events/MaterialNamesLoadEvent.java:8]: This is not useful imo. Other plugins can just always assume names to be loaded when their onEnable is called.

File: src/main/java/vg/civcraft/mc/civmodcore/CivModCorePlugin.java [hunk 4]
```diff
@@ -1,28 +1,52 @@
...
 		super.onEnable();
-		// needed for some of the apis
 		instance = this;
-		// we cant use the command annotation thing here, because the annotation
```
Comment (Maxopoly) [inline src/main/java/vg/civcraft/mc/civmodcore/CivModCorePlugin.java:14]: This comment should stay for the next person trying to fix this

**Review by Maxopoly — CHANGES_REQUESTED** [review]

File: src/main/java/vg/civcraft/mc/civmodcore/itemHandling/NiceNames.java (hunk 2 above)
Comment (Protonull) [reply src/main/java/vg/civcraft/mc/civmodcore/itemHandling/NiceNames.java:14]: NiceNames will be removed entirely within the next day with the addition of EnchantingAPI

Comment (Maxopoly) [reply src/main/java/vg/civcraft/mc/civmodcore/itemHandling/NiceNames.java:14]: Sounds good

File: src/main/java/vg/civcraft/mc/civmodcore/CivModCorePlugin.java (hunk 1 above)
Comment (Protonull) [reply src/main/java/vg/civcraft/mc/civmodcore/CivModCorePlugin.java]: The original code looks like so

```java
@Override
public void onEnable() {
	initApis(this);
}

private static synchronized void initApis(ACivMod instance) {
	if (!initializedAPIs) {
		// Initialises APIs
	}
}
```
whereby the API initialisation is synchronized. I figured it'd be best to continue that, though I'm not sure if onEnable and onDisable will be synchronized the same way that initApis was, but this is black magic to me, but making something synchronized makes it *more* thread safe, right?

Comment (Maxopoly) [reply src/main/java/vg/civcraft/mc/civmodcore/CivModCorePlugin.java]: No, adding more synchronized introduces the danger of dead locks, you cant just add more of them to make it safer. Just remove all synchronized here, it's not needed.

File: src/main/java/vg/civcraft/mc/civmodcore/events/MaterialNamesLoadEvent.java (hunk 3 above)
Comment (Protonull) [reply src/main/java/vg/civcraft/mc/civmodcore/events/MaterialNamesLoadEvent.java:8]: Huh, hadn't thought of that. But that would mean leave plugins unaware of when `ItemAPI.resetItemNames()` is called without the server starting or being reloaded.

File: src/main/java/vg/civcraft/mc/civmodcore/CivModCorePlugin.java (hunk 4 above)
Comment (Protonull) [reply src/main/java/vg/civcraft/mc/civmodcore/CivModCorePlugin.java:14]: Fair enough. I thought the comment specifying it must be done traditionally would be enough but if not, this is so minor this comment isn't really worth it :P

File: src/main/java/vg/civcraft/mc/civmodcore/CivModCorePlugin.java (hunk 1 above)
Comment (Protonull) [reply src/main/java/vg/civcraft/mc/civmodcore/CivModCorePlugin.java]: Fair enough

**Comment by Protonull** [comment]
Huh, force pushes are announced like that? Well just to be transparent, I reset the branch one commit because the command annotation comment didn't commit properly and didn't want to spam this page with a commit that is literally just a comment edit... buuuut, hey ho, lesson learnt.

**Pull request merged**